### ISSUE TITLE
bfs and soc-LiveJournal1: output should be quieted?

### ISSUE THREAD

**Issue by jowens** [gunrock/gunrock#101]
```
Starting  breadth_first_search soc-LiveJournal1
0	 3	 0	 queue3 	 oversize :	 4847573 ->	 6431004
0	 4	 0	 queue3 	 oversize :	 6431004 ->	 34759477
Ending  breadth_first_search soc-LiveJournal1
```

I don't know if this is an error or simply a printing message, but if it's the latter, it should be disabled by `--quiet`.

**Comment by sgpyc** [comment]
It's a printing message, by the array size checking mechanism. I wanted to disable it by --quiet, but the quiet_mode flag is not past down to the level that message is print. Shall I create a global variable (quite_mode, but I remember that, in general, using global variables is dis-encouraged, as their scope is not protected and vulnerable to hack / attack / mess up), or find a way to pass quiet_mode all to way down to the utilities functions?

**Comment by jowens** [comment]
I agree that global variables are bad. I went to a lot of effort last summer to distribute the quiet variable to all of the places it was needed. I guess find a way to pass it down the stack ...

**Comment by jowens** [comment]
Probably the better long-term solution is perhaps looking into logging packages for C++, but that's overkill for what we're currently doing.

**Comment by jowens** [comment]
Fixing this is not at all high priority. I mostly wanted to know nothing was wrong here.

Another option is to write the interesting queue-oversize info into the JSON (instead of to cout).

**Comment by sgpyc** [comment]
Maybe I can past a pointer to the Info object? It contains all possible flags, and @yzhwang and I have discussed this possibility. This can save the work next time when we need to pass another thing down.

**Comment by jowens** [comment]
Works for me.

**Comment by jowens** [comment]
Also PageRank:
```
Starting  pagerank soc-LiveJournal1
Reference total rank : 4845319.8761386424
Maximum difference : rank[8737] 6.82805115e+02 vs. 6.97301331e+02, 1.44962158e+01
Maximum relative difference :rank[1439348] 2.68825722e+00 vs. 2.83516026e+00, 5.18147200 %
Ending  pagerank soc-LiveJournal1
```

**Comment by neoblizz** [comment]
A new student can test this issue out for v1.x, we have a way to pass the quiet flag down the pipeline now. Will be a neat little project to understand gunrock's parameters support.
